cachi2/core/package_managers/pip.py

```diff
@@ -1477,38 +1481,41 @@ def _process_req(
 ) -> dict[str, Any]:
     download_info["kind"] = req.kind
     download_info["requirement_file"] = str(requirements_file.file_path.subpath_from_root)
-    download_info["hash_verified"] = False
+    download_info["missing_checksums"] = False
+    download_info["checksum_matched"] = False
+    # "package_type" is *only* needed for PyPI deps
+    download_info["package_type"] = ""
 
-    def _hash_verify(path: Path, checksum_info: Iterable[ChecksumInfo]) -> bool:
-        verified: bool = False
+    def _checksum_match(path: Path, checksum_info: Iterable[ChecksumInfo]) -> bool:
```

Comment:
Another super minor nitpicky suggestion. What do you think about something like the code below:
```suggestion
try:
           must_match_any_checksum(path, cheksum_info)
except PackageRejected:
            path.unlink()
            log.warning("Download '%s' was removed from the output directory", path.name)
else:
        return True
return False
```
`matched` is not really needed here, so could be removed completely.

Reply:
You'll find I'm *very* partial to the "single exit" philosophy. Locals hardly cost anything, and a single exit point is easier to debug and reason about.

Reply:
Fair enough. I would point out that in this specific case multiple exit points don't look like they would add extra complexity (they are right next to each other and highly unlikely to move further apart), but the published approach is totally fine. Thank you!

Reply:
Agreed on doing `return True` and `return False` here in this case, it's more direct and less confusing than using a variable.